modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/DLViewMVCRenderCommand.java

```diff
@@ -78,10 +80,23 @@ public String render(
 			renderRequest.setAttribute(
 				WebKeys.DOCUMENT_LIBRARY_FOLDER, _getFolder(renderRequest));
 
-			DLAdminDisplayContext dlAdminDisplayContext =
-				_dlAdminDisplayContextProvider.getDLAdminDisplayContext(
-					_portal.getHttpServletRequest(renderRequest),
-					_portal.getHttpServletResponse(renderResponse));
+			DLAdminDisplayContext dlAdminDisplayContext = null;
+
+			try {
+				dlAdminDisplayContext =
+					_dlAdminDisplayContextProvider.getDLAdminDisplayContext(
+						_portal.getHttpServletRequest(renderRequest),
+						_portal.getHttpServletResponse(renderResponse));
+			}
+			catch (SystemException systemException) {
+				if (systemException.getCause() instanceof
+						NoSuchFolderException) {
+
+					throw new NoSuchFolderException(systemException);
+				}
+
+				throw systemException;
+			}
```

Comment:
This is very weird. For what I've seen, the problem is that the `_computeFolders` method wraps the NSFE exn in a SystemException; why not fix that method instead? If you cannot make the thing throw a PortalException directly, it is better to use `ReflectionUtil.throw()` instead of wrapping the original exception.

Reply:
Thank you @adolfopa,

Unfortunately I don't think I can change to throwing the PortalException because `_computeFolders` is called on instantiation of DLAdminDisplayContext. I will try using `ReflectionUtil.throw()`.